drivers/watchdog/wdt_mcux_cop.c

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright (c) 2025 Alexandre Rey
+ *
+ * Based on wdt_mcux_rtwdog.c, which is:
+ * Copyright 2024, NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nxp_cop
+
+/**
+ * @brief Watchdog (WDT) Driver for NXP MCUX Computer Operating Properly (COP)
+ *
+ * Note:
+ * - Once the watchdog is disabled, it cannot be enabled till next
+ *   power reset, i.e, the watchdog cannot be started once stopped.
+ * - If the application does not require the watchdog to be used in the system,
+ *   CONFIG_WDT_DISABLE_AT_BOOT must be set in the application's config file.
+ */
+
+#include <zephyr/drivers/watchdog.h>
+#include <fsl_cop.h>
+
+#define LOG_LEVEL CONFIG_WDT_LOG_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(wdt_mcux_cop);
+
+struct mcux_cop_config {
+	SIM_Type *base;
+	cop_clock_source_t clk_source;
+	uint64_t timeout_cycles;
+	bool windowed_mode;
+};
+
+struct mcux_cop_data {
+	cop_config_t cop_config;
+	bool timeout_valid;
+	/* COP watchdog control register is write-once. setup_done tracks
+	 * whether the register has already been written.
+	 */
+	bool setup_done;
+};
+
+static int mcux_cop_setup(const struct device *dev, uint8_t options)
+{
+	const struct mcux_cop_config *config = dev->config;
+	struct mcux_cop_data *data = dev->data;
+
+	if (!data->timeout_valid) {
+		LOG_ERR("No valid timeouts installed");
+		return -EINVAL;
+	}
+
+	if (data->setup_done) {
+		LOG_ERR("Watchdog already setup, only one setup allowed");
+		return -EPERM;
+	}
+
+#if defined(FSL_FEATURE_COP_HAS_STOP_ENABLE) && FSL_FEATURE_COP_HAS_STOP_ENABLE
+	data->cop_config.enableStop = (options & WDT_OPT_PAUSE_IN_SLEEP) == 0U;
+#else
+	if ((options & WDT_OPT_PAUSE_IN_SLEEP) != 0U) {
+		LOG_ERR("WDT_OPT_PAUSE_IN_SLEEP not supported");
+		return -ENOTSUP;
+	}
+#endif
+
+#if defined(FSL_FEATURE_COP_HAS_DEBUG_ENABLE) && FSL_FEATURE_COP_HAS_DEBUG_ENABLE
+	data->cop_config.enableDebug = (options & WDT_OPT_PAUSE_HALTED_BY_DBG) == 0U;
+#else
+	if ((options & WDT_OPT_PAUSE_HALTED_BY_DBG) != 0U) {
+		LOG_ERR("WDT_OPT_PAUSE_HALTED_BY_DBG not supported");
+		return -ENOTSUP;
+	}
+#endif
+
+	data->setup_done = true;
+
+	COP_Init(config->base, &data->cop_config);
+	LOG_DBG("Setup the watchdog");
+
+	return 0;
+}
+
+static int mcux_cop_disable(const struct device *dev)
+{
+	const struct mcux_cop_config *config = dev->config;
+	struct mcux_cop_data *data = dev->data;
+
+	if (data->setup_done) {
+		LOG_ERR("Watchdog already setup, only one setup allowed");
+		return -EPERM;
+	}
+
+	data->timeout_valid = false;
+	data->setup_done = true;
```

Comment:
One concern here, if user did not enable COP and call `wdt_disable` before use it, do we really need to call `COP_Disable(config->base)` here. It may waste the write once chance.